tests/test_scraper.py

```
@@ -1,151 +1,365 @@
+
+@patch('requests.get')
+def test_scrape_url(mock_get, scraper, mock_response):
+    mock_get.return_value = mock_response
+    url = "https://example.com"
+    result = scraper.scrape_url(url, source="test")
+    assert result is not None
+    assert isinstance(result, list)
+    assert all(isinstance(item, PriceData) for item in result)
+
+def test_parse_response(scraper):
+    soup = BeautifulSoup(MOCK_HTML, 'html.parser')
```

Ruff (0.8.2)

Local variable `soup` is assigned to but never used

Remove assignment to unused variable `soup`

(F841)